packages/core/src/editor/model/Editor.ts

```diff
@@ -1136,4 +1137,60 @@ export default class EditorModel extends Model {
       el[varName][name] = value;
     }
   }
+
+  private sendTelemetryData() {
+    const sessionKeyPrefix = 'gjs_telemetry_sent_';
+
+    // @ts-ignore
+    const version = __GJS_VERSION__;
+    const sessionKey = `${sessionKeyPrefix}${version}`;
+
+    if (sessionStorage.getItem(sessionKey)) {
+      console.log(`Telemetry already sent for version ${version} this session`);
+      return;
+    }
+
+    // @ts-ignore
+    const url = __STUDIO_URL__;
+    const path = '/api/gjs/telemetry/collect';
+
+    (async () => {
+      const response = await fetch(`${url}${path}`, {
+        method: 'POST',
+        body: JSON.stringify({
+          type: 'EDITOR:LOAD',
+          domain: window.location.hostname,
+          version,
+          timestamp: Date.now(),
+          url,
+        }),
+      });
+
+      if (!response.ok) {
+        throw new Error(`Failed to send telemetry data ${await response.text()}`);
+      }
+
+      console.log(`Telemetry data sent successfully for version ${version}`);
+
+      sessionStorage.setItem(sessionKey, 'true');
+
+      Object.keys(sessionStorage).forEach((key) => {
+        if (key.startsWith(sessionKeyPrefix) && key !== sessionKey) {
+          sessionStorage.removeItem(key);
+        }
+      });
+
+      this.trigger('telemetry:sent');
+    })().catch((err) => {
+      console.error('Failed to send telemetry data', err);
+    });
+  }
+
+  trackEditorLoad() {
+    this.on('load', () => {
+      if (this._config.telemetry) {
+        this.sendTelemetryData();
+      }
+    });
+  }
```

Comment:
@artf what is the best way to hook into editor load, should we do it on the main editor instance or the model, and what event is appropriate for the "EDITOR:LOAD" event tracking?

Reply:
Let's trigger `sendTelemetryData` from [here](https://github.com/GrapesJS/grapesjs/blob/d74c3f18a873bac5bd9658659932acf165c91ef5/packages/core/src/editor/view/EditorView.ts#L10-L17). `EDITOR:LOAD` sounds good to me